packages/brisa/src/utils/rpc/rpc-integration.test.ts

```diff
@@ -395,6 +396,9 @@ describe('utils', () => {
         navigationType?: 'push' | 'replace';
       } = {},
     ) {
+      if (location.href === url) {
```

Comment:
This condition logic is already inside the RPC that we want to test. I mean adding `preventDefault` on line 420, a similar mock strategy than `mockNavigationIntercept`

Reply:
I tried adding `preventDefault` to the dummy `fn` event like so:

```js
    const preventDefault = mock(() => {});

    fn({
        preventDefault,
        destination: { url },
        scroll: () => {},
        ...
```

However, it seems that this does not trigger the `event.preventDefault` method. Is there something I'm missing ?

Reply:
I guess It's because inside the `simulateSPANavigation` we have this on line 408:

```js
location.href = origin;
```

We need to adapt the simulator and we should be able to pass the `locationHref` in the configuration and do something like this:

```
location.href = locationHref ?? origin;
```

Then:

```js
const url = 'http://localhost/some-page';
await simulateSPANavigation(url, { locationHref: url });
```

Probably something like that should adapt the simulator.